sdk/maps/maps-search/src/mapsSearchClient.ts

```diff
@@ -728,48 +723,60 @@ export class MapsSearchClient {
    * rehydratedPoller.poll()
    * ```
    *
-   * @param resumeFrom - The serialized state from the previous poller.
+   * @param restoreFrom - The serialized state from the previous poller.
    * @param options - Optional parameters for the operation.
    *
    */
   public async resumeSearchAddressBatch(
-    resumeFrom: string,
+    restoreFrom: string,
     options: SearchAddressBatchOptions = {}
-  ): Promise<BatchPoller<BatchResult<SearchAddressResult>>> {
-    return this.createSearchAddressBatchPoller(undefined, { ...options, resumeFrom });
+  ): Promise<SearchAddressBatchPoller> {
+    return this.createSearchAddressBatchPoller(restoreFrom, options);
   }
 
   private async createSearchAddressBatchPoller(
-    requests: SearchAddressRequest[] = [],
-    options: SearchSearchAddressBatchOptionalParams = {}
-  ): Promise<BatchPoller<BatchResult<SearchAddressResult>>> {
-    const { span, updatedOptions } = createSpan(
-      "MapsSearchClient-beginSearchAddressBatch",
-      options
+    requestsOrRestoreFrom: SearchAddressRequest[] | string,
+    options: SearchAddressBatchOptions = {}
+  ): Promise<SearchAddressBatchPoller> {
+    return this.tracing.withSpan(
+      "MapsSearchClient.beginSearchAddressBatch",
+      options,
+      async (updatedOptions) => {
+        const requests = Array.isArray(requestsOrRestoreFrom) ? requestsOrRestoreFrom : [];
+        const restoreFrom = typeof requestsOrRestoreFrom === "string" ? requestsOrRestoreFrom : "";
+        const batchRequest = createSearchAddressBatchRequest(requests);
+
+        const poller = await createHttpPoller(
+          {
+            sendInitialRequest: async () => {
+              return getRawResponse(
+                async (paramOptions) =>
+                  this.client.search.searchAddressBatch(
+                    this.defaultFormat,
+                    batchRequest,
+                    paramOptions
+                  ),
+                updatedOptions
+              );
+            },
+            sendPollRequest: createSendPollRequest({
+              client: this.client,
+              options: updatedOptions,
```

Comment:
are the options the same for the initial request and polling requests?

Reply:
yes, they are the same.